ranker.py

```diff
@@ -0,0 +1,63 @@
+import string
+from utils import *
+
+IDEAL_Q_LENGTH = 10
+MAX_SCORE = 17
+
+def sortSecond(val):
+    return val[1]
+
+class Ranker:
+    '''
+    Holds the ranking data structure for how questions are scored and sorted
+    The q_list becomes a max priority queue, with the max score at the front.
+
+    Properties:
+        q_list: (list(Question, int)) list tuples of unranked questions and scores
+        avg_coref_len: (int) average length of coref clusters
+    '''
+    
+    def __init__(self, question_list, avg_coref_len):
+        self.q_list = [(question, 0) for question in question_list]
+        # self.doc = doc
+        self.avg_coref_len = avg_coref_len
+
+    def _rank_question(self, question):
+        # is entity: +10
+            # is main ref: + 5
+                # ent size is large: +2
+        # -abs(dist from 10 words) / 3
+        score = 0
+        question_ents = question.entities()
+        if len(question_ents) > 0:
+            score += 10
+            for ent in question_ents:
+                if ent == ent._.coref_cluster.main:
+                    score += 5
+                    if len(ent._.coref_cluster) > self.avg_coref_len:
+                        score += 2
+                    break
+        q_len = len(question.q_string.split())
```

Comment:
We COULD pass the question string back into spaCy and re-process it (maybe giving Ranker the instance of Preprocessor that exists in 'ask'). 

If we did that, we could easily penalize punctuation like commas or longer/multiple clauses (which may indicate questions that are too specific or complicated)